Firmware/fibre/cpp/include/fibre/protocol.hpp

```diff
@@ -882,37 +883,48 @@ class ProtocolProperty : public Endpoint {
     }
     void handle(const uint8_t* input, size_t input_length, StreamSink* output) final {
         default_readwrite_endpoint_handler<TProperty>(property_, input, input_length, output);
+        if (written_hook_ != nullptr && input_length >= sizeof(TProperty)) {
```

Comment:
Although currently `input_length >= sizeof(TProperty)` is _functionally_ a necessary condition that the property was changed, _conceptually_ it is not. Meaning if you change the on-the-wire encoding this check might no longer be correct. It might be safer to have `default_readwrite_endpoint_handler` return a boolean and use that.

Reply:
Yeah it felt dirty, but I didn't see another way to do it in this scope. The return value is a great idea, I'll do that instead.